lib/Optimizer/CodeGen/Pipelines.cpp

```diff
@@ -45,20 +45,11 @@ void cudaq::opt::addPipelineTranslateToOpenQASM(PassManager &pm) {
   pm.addPass(createCanonicalizerPass());
   pm.addPass(createCSEPass());
   pm.addNestedPass<func::FuncOp>(createClassicalMemToReg());
-  pm.addPass(createLoopUnroll());
-  pm.addPass(createCanonicalizerPass());
-  pm.addNestedPass<func::FuncOp>(createLiftArrayAlloc());
-  pm.addPass(createGlobalizeArrayValues());
-  pm.addPass(createStatePreparation());
-  pm.addNestedPass<func::FuncOp>(createGetConcreteMatrix());
-  pm.addPass(createUnitarySynthesis());
+  pm.addNestedPass<func::FuncOp>(createExpandControlVeqs());
+  pm.addNestedPass<func::FuncOp>(createCombineQuantumAllocations());
+  pm.addNestedPass<func::FuncOp>(createCombineMeasurements());
```

Comment:
Does using OpenQASM2 require measurements to be replaced with "measure everything and sort it out later"?

Reply:
Yes, I am following a similar path to what we do in Ionq: measure everything, keep a mapping for the qubits we want to report, and filter out counts after the execution.

Reply:
moved the extra passes to python's code.